Documentation/building/linux-instructions.md

```diff
@@ -158,3 +158,63 @@ Finally, let's build coreclr with updated clang/llvm. If you meet a lldb related
 lgs@ubuntu time ROOTFS_DIR=/work/dotnet/rootfs-coreclr/arm ./build.sh arm release clean cross 
 ```
 
+Additional optimization levels for ARM/Linux: -Oz and -Ofast
+============================================================
+
+This instruction is to enable -Oz or -Ofast optimization level of clang on ARM/Linux. There are the optimization levels of clang/LLVM that have similar options with optimization flags of Microsoft Visual Studio 2015 compiler. The below table shows what we have to enable for the code optimization of the CoreCLR run-time either the size or speed on embedded devices. 
+
+| **Content** | **Visual Studio (Windows)** | **Clang/LLVM (Linux)** |
+| --- | --- | --- |
+|To Disable Optimization| /Od | -O0 (debug build)|
+|To Minimize Size| /O1 | -Oz |
+|To Maximize Speed| /O2  | -Ofast |
+|To Support Full Optimization| /Ox | -O3 (release build) |
```

Comment:
> What does "Full Optimization" mean? I would remove this.

When I checked the Ox flag from MS Visual Studio 2015, I thought that the meaning of the "Full Optimization" means  optimization considering both speed and size. At that time, I could not find the exact meaning at https://msdn.microsoft.com/en-us/library/k1ack8f1.aspx

>  I would remove this.

I made a mistake. At that time I did not change the configuration of VS2015 from Debug to Release. I will remove that. 

> O3 and Ofast are very much the same in Clang. 
> Also, I believe that Ox is actually less optimization than O2 in Microsoft compilers.

https://www.freebsd.org/cgi/man.cgi?query=clang
  -Ofast:      Enables all the optimizations from -O3 along with other aggressive optimizations	that may violate strict	compliance with language standards.